### ISSUE TITLE
Support [ci skip]

### ISSUE THREAD

**Issue by burnettk** [Eernie/bitbucket-webhooks-plugin#49]
If the term "[ci skip]" appears in the commit message, 1) don't send any webhooks and 2) profit.

Idea from https://stackoverflow.com/questions/44324122/ci-skip-not-working-with-bitbucket. This is a github/travis ci feature (and implemented in various combinations of CI systems and source control hosting solutions) that is quite possibly the greatest feature ever devised. Am I right? :)

**Comment by casz** [comment]
I take it, this would only apply to a single commit to a branch? not PR events (that is not commit event) or merges?

**Comment by burnettk** [comment]
@casz that sounds right. so if you have an outstanding pull request for a branch when you make a commit with [ci skip], it might still send the PR event, if I'm understanding correctly. we're waiting with bated breath for the "make configurable which branches and PRs get built" jenkins feature that stephen connolly (sp?) is working on. we don't ever want to build PRs, since they duplicate feature branch builds. i think it may be ok/desirable to ignore merges for this change as well, like you say.

thoughts about whether this is a good idea or how difficult it might be to implement are welcome!

**Comment by casz** [comment]
Perhaps this is better to be implemented over at bitbucket https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/20
To at least show that it was skipped on Jenkins 👍

**Comment by burnettk** [comment]
Yes, I do think it would be slightly better implemented in jenkins, since then it could work for all webhooks and even other use-cases. It also feels slightly harder, so if that pull request gets closed as "won't fix" / "don't care" i may re-open this one. :) thanks much for the pointer!

**Comment by casz** [comment]
@burnettk https://plugins.jenkins.io/bitbucket-scm-trait-commit-skip :)

**Comment by burnettk** [comment]
@casz super exciting!

**Comment by marbon87** [comment]
Hi, i was excited about the bitbucket-scm-trait-commit-skip-plugin but then i realized that it only works for skipping the builds of pull request containt '[ci skip]'.
But we want so skip every build on every branch in jenkins, e.g. to skip builds for maven release commits.

After reading https://issues.jenkins-ci.org/browse/JENKINS-45502 and https://issues.jenkins-ci.org/browse/JENKINS-47643 it seems that it is not easy to implement this using a jenkins plugin, so maybe it is still a good idea to implement this feature in the webhook plugin? what do you think?

**Comment by burnettk** [comment]
@marbon87 thanks for checking into this. i would still support adding it on the bitbucket server side, since the jenkins side seems like it may never happen, and this is a useful feature.

in our case, if bitbucket server stopped sending webhook events in this case, i think it would still cause confusion, since we have jenkins building repos every day even if the repo doesn't get a webhook event (to make sure it doesn't miss webhook calls, etc), so it wouldn't be perfect, but it would be better than the status quo, i think.

**Comment by casz** [comment]
@marbon87 & @burnettk a PR would be great 😌
